src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -756,8 +756,6 @@ def _prepare_decoder_attention_mask(self, attention_mask, input_shape, inputs_em
             ).to(inputs_embeds.device)
 
         if attention_mask is not None:
-            if attention_mask.shape[-1] > input_shape[-1] > 0:
-                attention_mask = attention_mask[:, : input_shape[-1] + past_key_values_length]
```

Comment:
This condition was there since the model addition PR.
However, I think if `attention_mask` has longer length than `input_shape` length, it **should** have the length `input_shape[-1] + past_key_values_length` rather than being longer than this. Otherwise, it means something very strange happens, and we should either catch it and throw explicit error, or leave later ops to fail.

Reply:
I think I had to add it at some point, but then it became irrelevant. Thanks for this I guess!